crates/api/src/community/add_mod.rs

```diff
@@ -29,7 +29,7 @@ pub async fn add_mod_to_community(
   is_mod_or_admin(&mut context.pool(), local_user_view.person.id, community_id).await?;
   let community = Community::read(&mut context.pool(), community_id).await?;
   if local_user_view.local_user.admin && !community.local {
-    return Err(LemmyErrorType::NotAModerator)?;
+    Err(LemmyErrorType::NotAModerator)?
```

Comment:
I finished going through the codebase, and removed most of the `return Err ... ; ` and changed them to `Err ... (no semicolon)` where possible. If you'd like, I can move that to a different PR.

Reply:
This kind of code will be added right back if we dont prevent it via clippy. Closest I could find in this regard is [try_err](https://rust-lang.github.io/rust-clippy/stable/index.html#try_err) which only allows the way with return.

Reply:
Actually I think I change my mind on this one. According to [this lint](https://rust-lang.github.io/rust-clippy/stable/index.html#needless_return), and [this lint](https://rust-lang.github.io/rust-clippy/stable/index.html#implicit_return) omitting the return is considered more idiomatic rust.

Reply:
I also ran the `try_err` --fix just to test it, and it conflicts with the `needless_return` lint, which then throws a ton of warnings.

Reply:
I'll merge this as is then, since I did a lot of manual work to make those returns more logical.